qiita_pet/uimodules/prep_template_tab.py

```diff
@@ -0,0 +1,274 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from operator import itemgetter
+from os.path import basename
+from collections import defaultdict
+
+from future.utils import viewitems
+
+from qiita_db.util import (get_filetypes, get_files_from_uploads_folders,
+                           get_data_types, convert_to_id, get_filepath_types)
+from qiita_db.study import Study
+from qiita_db.data import RawData
+from qiita_db.ontology import Ontology
+from qiita_db.metadata_template import (PrepTemplate, TARGET_GENE_DATA_TYPES,
+                                        PREP_TEMPLATE_COLUMNS_TARGET_GENE)
+from qiita_db.parameters import (Preprocessed454Params,
+                                 PreprocessedIlluminaParams)
+from qiita_pet.util import STATUS_STYLER
+from qiita_pet.handlers.util import download_link_or_path
+from .base_uimodule import BaseUIModule
+
+
+filepath_types = [k.split('_', 1)[1].replace('_', ' ')
+                  for k in get_filepath_types()
+                  if k.startswith('raw_')]
+fp_type_by_ft = defaultdict(
+    lambda: filepath_types, SFF=['sff'], FASTA=['fasta', 'qual'],
+    FASTQ=['barcodes', 'forward seqs', 'reverse seqs'])
+
+
+def _get_accessible_raw_data(user):
```

Comment:
Another thing ... there's no tests for these functions, which I think we are ok as this is a transitive step, but I wanted to highlight this in case someone else has stronger feelings about it.